tests/lib/rules/no-unused-prop-types.js

```diff
@@ -784,54 +764,6 @@ ruleTester.run('no-unused-prop-types', rule, {
       ].join('\n')
     }, {
       code: [
-        'const statelessComponent = (props) => {',
```

Comment:
If you ask me, the fact that certain cases are not supported could also be tested. Of course, with a test name which clearly states "testing unsupported case" and assertions that clearly check for false positives or other signs of "not supported"-ness.